### DISCUSSION TITLE
changing medium voltage input

### DISCUSSION THREAD

**Discussion by willemijnbrus** [PowerGridModel/.github#30] [Q&A] [answered]
Hi PGM team,

With my team I am writing a model to get insights in the flex control on the low voltage level part of the grid. 
We implement flex control in ev, wp and pv profiles and see what the consequences are (can we avoid congestion given the contstraints, what is the impact on the  profiles etc). 
We are also monitoring the voltage. 

We assume symmetric loadflow while in practice that is not the case. 
A later expansion of the model will be to randomnize the phase heatpumps, houses are connected on and see what it will do with the voltage. 
Moreover, we are aware that the tap positions of the transformer that we have in our data are not always correct (research collaboration with silo on this topic).
Later on, it will be interesting if we can use some of the results of the above research to improve our data quality. 

At the moment, we see that the results always give relatively high voltages (min voltage over all nodes over the year is around 240 and maximum is around 255 Volt). We see this for more runs for different msr grids. 
Therefore, we were wondering if it is possible that our data give a high medium voltage (10.5kv rather than 10kv) which causes a relatively high lower voltage value lower in the grid. 
To check this, I checked transformer["u1"]. As a quick fix, we are thinking about lowering the medium voltage with 0.5kV and see if the voltage in the nodes in the lower voltage part of the grid will be lower. 
However, I am doubting if I could do that by changing transformer["u1"] or by changing something in the source. 
I do not fully understand the role of the transformer vs the source. Specifically what the reference voltage in per-unit exactly means and if I need this. 

**My questions to you are:
How are you thinking about changing the medium voltage with 0.5kV as a quick fix?
Should I do this via transformer[u1]?
What is exactly the role of a transformer vs source?
What does exactly reference voltage per unit mean?**

I tried to look it up in the documentation but I could not find the answer to my questions. 
I hope you have time to help me 😊

**Comment by petersalemink95** [comment]
Hi @willemijnbrus,

@nitbharambe and @zhen0427 will schedule a meeting to discuss this in detail.
First thoughts: the proposed quick fix does not feel like the right way forward. Probably tap settings need to be adjusted.

p.s. the experimental feature release of automatic tap changer is ready -> this might help you :)

**Reply by willemijnbrus** [reply]
Hi Peter,
Thanks for your answer! Great that nitish bharambe and zhen wang, are willing to have a meeting on this matter.
I have red something in the discussion about the automatic tap changing, I will have another look at it.

**Reply by petersalemink95** [reply]
The final release is not yet ready, but it can already be used as experimental feature. @nitbharambe and @zhen0427 can explain how, but you can also rewatch our last sprint review.

**Reply by mgovers** [reply]
Hi @willemijnbrus ,

We released an example notebook in https://power-grid-model.readthedocs.io/en/stable/examples/Transformer%20Examples.html#power-flow-calculations-with-automatic-tap-changing that illustrates how to use the automatic tap changing power flow calculations, even if they are still experimental. Documentation will be available in the coming hours (again with the caveat that it's still experimental).

**Reply by willemijnbrus** [reply]
thanks martijn!

**Reply by nitbharambe** [reply, marked as answer]
The temporary conclusion is to see if tap changer regulators are indeed present in the grid and implement them per the example notebook as mentioned by Martijn. And if there is something more needed than that we shall have to do a deeper dive.

The questions you ask additionally are good power engineering refreshers. Although all information might not be relevant, its nice to address them here:

1) Changing the `u_rated` is possible only for power flow calculations. And its effects are only all  the `p.u.` results. It does not affect the real unit results like `V`, `A`, `W`, `Var`, `VA`. If the user intends to do it, it should be done with a good cognizance of its effects. **However this generally not recommended to do so.**

2) Changing `u1` wont be accurate as those corresponds to actual rating of transformer. ie, the internal impedance of transformer would be affected by it.

3) The transformer converts a voltage level to another level. It has a `pi` like impedance model in PGM and the voltage is multiplied by a ratio corresponding to `u1` and `u2` across the nodes. 
A source is a component where the voltage is guaranteed to be equal to `u_ref` p.u. This is called a `slack bus` in load flow theory. A minimum of one source is required for all power flow / state estimation calculations. It is where you shall find all incoming power to grid (ie. including all grid losses) being attributed to. In context of a DSO this is almost equivalent to a connection with the TSO. 

4) `p.u.` means per unit or percent. In context of all voltages in PGM, it is percent of the `u_rated` voltages specified at nodes. This is more of a convention and it helped with manual calculations in past. It does not affect the real units of `V`, `A`, `W`, `Var`, `VA`. As the grid is composed of equipment with voltages as high as 100s of kV and as low as 400V, it is more convenient to compare with p.u. instead. A grid operator for example expects voltages to be in neighborhood of 1 p.u.

**Reply by willemijnbrus** [reply]
Thank you for your answer!

**Reply by mgovers** [reply]
FYI: power-grid-model version [v1.8.0](https://github.com/PowerGridModel/power-grid-model/releases/tag/v1.8.0) was released yesterday, which, among other things, contains support for power flow calculations with automatic tap adjustment. Feel free to try it out and give feedback. We're happy with any input.

Please also feel free to mark this as answer if it indeed solves your issues.
